File: Punto_4/mcd.hs
import System.CPUTime
import Text.Printf

mcd :: Int -> Int -> Int
mcd a 0 = a
mcd a b = mcd b (mod a b)

timeFunc :: IO a -> IO a
timeFunc action = do
    start  <- getCPUTime
    result <- action
    end    <- getCPUTime
    let diff = fromIntegral (end - start) / (10^12) :: Double
    printf "Tiempo de ejecución: %0.6f segundos\n" diff
    return result

main :: IO ()
main = do
    linea <- getLine
    let [a, b] = map read (words linea) :: [Int]
    resultado <- timeFunc $ do
        let r = mcd a b
        print r        -- fuerza evaluación lazy
        return r
    putStrLn $ "MCD(" ++ show a ++ ", " ++ show b ++ ") = " ++ show resultado

File: Punto_4/mcd.c
#include <stdio.h>
#include <time.h>

int mcd(int a, int b) {
    if (b == 0) return a;
    return mcd(b, a % b);
}

int main() {
    int a, b;
    scanf("%d %d", &a, &b);

    clock_t inicio = clock();
    int resultado = mcd(a, b);
    clock_t fin = clock();

    printf("MCD(%d, %d) = %d\n", a, b, resultado);
    printf("Tiempo: %f segundos\n", (double)(fin - inicio) / CLOCKS_PER_SEC);
    return 0;
}
